app/services/katello/pulp3/repository.rb

```diff
@@ -45,8 +45,22 @@ def distribution_options
         fail NotImplementedError
       end
 
+      def distribution_mirror_options(path, options = {})
+        ret  = {
+            base_path: path,
+            name: "#{backend_object_name}"
+        }
+        ret[:publication] = options[:publication] if options.key? :publication
+        ret[:repository_version] = options[:repository_version] if options.key? :repository_version
+        ret
+      end
```

Comment:
if you could move this lower in the file, the intent was to keep all the method that simpley `fail NotImplementedError`  together